src/lint/lint.nim

```diff
@@ -18,7 +18,23 @@ proc isValidTrackConfig(trackDir: string): bool =
   else:
     writeError("Missing file", configJsonPath)
 
+proc subdirsContain(dir: string, files: openArray[string]): bool =
+  ## Returns `true` if every file in `files` exists in every subdirectory of
+  ## `dir`.
+  ##
+  ## Returns `true` if `dir` does not exist or has no subdirectories.
+  result = true
+
+  if dirExists(dir):
+    for subdir in getSortedSubdirs(dir):
+      for file in files:
+        let path = subdir / file
+        if not fileExists(path):
+          writeError("Missing file", path)
```

Comment:
Should this be:

```suggestion
          writeError("Missing file", path)
          result = false
```

Reply:
No, because the current `writeError` is:
https://github.com/exercism/configlet/blob/a7f090efeef1409b7edb7c430a474445175774f2/src/lint/lint.nim#L4-L8

Due to how templates work, `writeError` is already setting the `result` of the `subdirsContain` proc to `false`. 

Try running `configlet lint` with this PR using some track dir that has missing files. You should see that we detect them, but the only place that can set `result` to `false` is inside the `writeError` template.

I might refactor this pattern later though. I haven't settled on the most expressive/maintainable way to write the linting code, but this is one alternative to writing something like:
```Nim
proc isValidConfigFile(...): bool =
  result = true
  if not isValidFoo(...):
    result = false
  if not isValidBar(...):
    result = false
```
everywhere, or passing `result` as a `var` parameter to each validator function.

See the use of `writeError` in https://github.com/exercism/configlet/pull/169/ - in particular, how the `result` of `isValidConceptExerciseConfig` can become `false`. I don't claim this is the best approach, though.

We could probably improve the `writeError` name - any ideas?

Reply:
Ah, now I understand. I think the original pattern is fine, but maybe with some rename as suggested. But what to use as the name... 🤔 The only thing I could come up with is `handleError`, but I don't know.